agent/status.go

```diff
@@ -1,31 +1,36 @@
 package agent
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"net/http"
 	"os"
+	"time"
 
 	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/service"
 	"github.com/gorilla/mux"
 	"github.com/mongodb/grip"
 	"github.com/mongodb/grip/message"
 	"github.com/urfave/negroni"
 )
 
-func (agt *Agent) startStatusServer(port int) {
+func (agt *Agent) startStatusServer(ctx context.Context, port int) {
 	addr := fmt.Sprintf("127.0.0.1:%d", port)
+	r := mux.NewRouter().StrictSlash(false)
+	r.HandleFunc("/status", agt.statusHandler()).Methods("GET")
+	r.HandleFunc("/terminate", terminateAgentHandler).Methods("DELETE")
 
-	go func() {
-		r := mux.NewRouter().StrictSlash(false)
-		r.HandleFunc("/status", agt.statusHandler()).Methods("GET")
-		r.HandleFunc("/terminate", terminateAgentHandler).Methods("DELETE")
-
-		n := negroni.New()
-		n.Use(negroni.NewRecovery())
-		n.UseHandler(r)
+	n := negroni.New()
+	n.Use(negroni.NewRecovery())
+	n.UseHandler(r)
 
-		grip.CatchEmergencyFatal(http.ListenAndServe(addr, n))
+	go func() {
+		err := service.RunGracefully(addr, 10*time.Second, n)
+		if err != nil {
+			grip.Notice(err.Error())
+		}
```

Comment:
grip methods handle errors correctly, so you can just: 

grip.Notice(service.RunGracefully(addr, 10*time.Second, n))